Kwc/Directories/List/Component.php

```diff
@@ -25,6 +25,10 @@ public static function getSettings()
             if (!is_instance_of($c, 'Kwc_Directories_Item_DirectoryNoAdmin_Component')) {
                 throw new Kwf_Exception("_getItemDirectory must return an Kwc_Directories_Item_DirectoryNoAdmin_Component data object or class-name, '{$c}' given; componentClass is ".get_class($this));
             }
+            $isData = call_user_func(array(get_class($this), 'getItemDirectoryIsData'), $this->getData()->componentClass);
+            if (is_string($this->_itemDirectory) && $isData) {
+                throw new Kwf_Exception("_getItemDirectory returns string, so getItemDirectoryIsData must return false; componentClass is ".get_class($this));
+            }
```

Comment:
Isnt' the opposite missing?
if ($this->_itemDirectory instance of Kwf_Component_Data && !$isData)